### ISSUE TITLE
Artist list limited on Android Auto

### ISSUE THREAD

**Issue by theksepyro** [VinylMusicPlayer/VinylMusicPlayer#890]
**Describe the bug**
The full list of artists (or albums) is no longer available when using the app through Android Auto.

**To reproduce**
1. Connect device to Android Auto
2. Navigate to Vinyl App
3. Scroll down to bottom of available artists
4. See "Limited listing while driving, Open app to see the full list"

**Expected behavior**
Like in previous versions of the app, I expect to be capable of viewing and selecting from the full list of artists in the library.

**Device info**
App version: 1.6.2
App version code: 188
Android build version: 11049176
Android release version: 14
Android SDK version: 34
Android build ID: UQ1A.231205.014
Device brand: google
Device manufacturer: Google
Device name: barbet
Device model: Pixel 5a
Device product name: barbet
Device hardware name: barbet
ABIs: [arm64-v8a, armeabi-v7a, armeabi]
ABIs (32bit): [armeabi-v7a, armeabi]
ABIs (64bit): [arm64-v8a]

**Additional context**
The change doesn't appear to be related to android auto itself since when reinstalling earlier versions of the app the full list becomes available again. 
The value in Android Auto is the safety of not having to pick up the device to change music while driving but this subverts that

**Comment by theksepyro** [comment]
I've been playing with it a little more and thinking about why it might be happening. I don't have any background in app development whatsoever though so this is largely speculation.

In the android auto interface previously, when selecting an artist in the artist tab, the first track of the first album would begin playing immediately. 

Now when selecting an artist, another window comes up that allows you to select an album by that artist, and then when you've selected one another still to select a song from that album. 

This additional granularity in selection is really nice from a usability standpoint, but I think it's possibly making it such that there are exponentially more "entries" in the artist tab or something. 

I can envision that android auto might have some maximum number of entries allowed in that screen and that before when only artists were selectable I was below that threshold and therefore I could see everything, but now that artist, albums, and songs entries are all available from the artists tab, I've gone way over that threshold.

For me personally, even though I really like the ability to choose songs on a per album basis, and albums on a per song basis, this is not worth sacrificing the ability to listen to most of my library without picking up my phone.

**Comment by soncaokim** [comment]
Hello,
The change you described has been made recently, mostly to improve responsiveness on old devices.

Previously the whole artist list is loaded, and if you have 1000 ones it just freezes. In addition to album browse, I've put the listing limit to every listing in the app' Auto interface.

This listing limit existed, was at 100 elements, and now I've reduced it to 50. If you have more elements to browse, that's dangerous to continue in Auto mode. The intention is make you to postpone what you are doing, not picking up your phone.

**Comment by theksepyro** [comment]
So if I'm understanding properly, no longer being able to access my full library through android auto is not a bug, but an intended change.

As far as what's dangerous to do in Android Auto mode, there is already a built in feature that stops you if you are scrolling too much and tells you effectively "scrolling is paused, watch the road instead of the screen." 

The primary way that I personally browse artists is by pressing the "a-z" button and then selecting the first letter of the artist before doing any scrolling, but now I can only select a-g. 

So there is functionally no difference in how I use the interface other than not being able to select h-z.

**Comment by theksepyro** [comment]
Seeing as it's intentional and therefore not a bug per se, I'll close this. 

That said, and I hope you do not take me as being combative or hostile or something, I find this change frustrating and will likely just build the app myself with the limit increased from here on. I can appreciate the argument about better supporting old devices, but hope that at some point the developers change their minds.

Thanks for the work you do.

**Comment by soncaokim** [comment]
Hi there,

Sorry I did not reply earlier. No hostility taken.

In fact we can consider having the listing limit configurable (i.e 50, 100, 200,...). I was not aware that Auto has an anti-scroll feature, you described - probably this has been added after this list limiting has been implemented with this app (months or years ago).

If you consider make a change in the code, I can point out where it's done and will conder merging this.

S.
